pkg/asset/manifests/scheduler.go

```diff
@@ -44,7 +48,29 @@ func (s *Scheduler) Generate(dependencies asset.Parents) error {
 			Name: "cluster",
 			// not namespaced
 		},
-		Spec: configv1.SchedulerSpec{},
+		Spec: configv1.SchedulerSpec{
+			MastersSchedulable: false,
+		},
+	}
+
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
+	computeReplicas := int64(0)
+	for _, pool := range installConfig.Config.Compute {
+		if pool.Replicas != nil {
+			computeReplicas += *pool.Replicas
+		}
+	}
+	if computeReplicas == 0 {
+		// A schedulable host is required for a successful install to complete.
+		// If the install config has 0 replicas for compute hosts, it's one of two cases:
+		//   1. An IPI deployment with no compute hosts.  The deployment can not succeed
+		//      without MastersSchedulable = true.
+		//   2. A UPI deployment.  The deployment may add compute hosts, but to ensure the
+		//      the highest probability of a successful deployment, we default to
+		//      schedulable masters.
+		logrus.Debugf("Defaulting to MastersSchedulable = true in Scheduler config.")
```

Comment:
Would a message at the INFO level be better?

Reply:
Yes please.

Reply:
done